src/pages/[productSlug]/tutorials/[...tutorialSlug]/index.tsx

```diff
@@ -5,20 +5,53 @@ import {
 } from 'next'
 import { LearnProductData, LearnProductSlug } from 'types/products'
 import { cachedGetProductData } from 'lib/get-product-data'
+import { getStaticPathsFromAnalytics } from 'lib/get-static-paths-from-analytics'
 import TutorialView from 'views/tutorial-view'
 import {
 	TutorialPageProps,
 	TutorialPagePaths,
 	getTutorialPagePaths,
 	getTutorialPageProps,
 } from 'views/tutorial-view/server'
+import { activeProductSlugs } from 'lib/products'
 
 async function getStaticPaths(): Promise<
 	GetStaticPathsResult<TutorialPagePaths['params']>
 > {
-	const paths = await getTutorialPagePaths()
+	const validPaths = await getTutorialPagePaths()
+
+	let paths = []
+
+	try {
+		paths = (
+			await Promise.all(
+				activeProductSlugs.map(async (productSlug) => {
+					// fetch paths from analytics for each product
+					const analyticsPaths = await getStaticPathsFromAnalytics<
+						TutorialPagePaths['params']
+					>({
+						param: 'tutorialSlug',
+						limit: __config.learn.max_static_paths ?? 0,
+						pathPrefix: `/${productSlug}/tutorials`,
+						validPaths,
+					})
+
+					// add the productSlug param to the resulting params object
+					return analyticsPaths.map((result) => {
+						result.params.productSlug = productSlug
+
+						return result
+					})
+				})
+			)
+		).flat()
+	} catch {
+		// In the case of an error, fallback to using the base list of generated paths to ensure we do _some_ form of static generation
+		paths = validPaths.slice(0, __config.learn.max_static_paths ?? 0)
+	}
```

Comment:
This is a little funky, but due to the dynamic `productSlug` path portion we've gotta do some trickery. The `validPaths` filtering still works because we ignore the `productSlug` path segment.

In the future, I could see us expanding the `static_paths` endpoint to support multiple levels of params, but this works for now!